### ISSUE TITLE
BBIM - Bug?- IFC4x3 PredefinedType vs ObjectType of IfcAnnotation

### ISSUE THREAD

**Issue by jes-r** [IfcOpenShell/IfcOpenShell#5314]
Hi,

When I'm in a project with the IFC4x3 schema, an issue shows when placing annotations. PredefinedType of annotations is filled in on placement (this option doesn't show in IFC4x0 projects). But to give the symbol, text or whatever it's options like being able to put in text, you need to fill in ObjectType (which is filled in by default in 4x0). When you try to type in eg. 'TEXT' it gives an error, but not always.

For this issue I tried recreating the problem on purpose but didn't get the error doing exactly the same but a second time. However I had this error show up 2 days ago too doing the same things.

Steps:
1.  Add an annotation, in this example Text. Typed or untyped doesn't matter.
2.  Notice how text gets placed, but no 'Text' word which should be there
3.  Only the PredefinedType is visible and filled in
4.  The annotation gets dropped in a new Unsorted collection **too**, is this how it should be?
5. I have to type in TEXT in ObjectType to give the proper settings
6. I can still change PredefinedType to eg. LEADER. Is this on purpose?
7. Saving in this case worked, but not always. Now that TEXT is added to ObjectType, PredefinedType changes to USERDEFINED?
8.  The 'TEXT' word is added and I can now put in the text. This is working, but seems 

![image](https://github.com/user-attachments/assets/2f42b1d5-f9b5-4a96-9928-8cf324c244d9)
![image](https://github.com/user-attachments/assets/08694b4b-a7f1-4a3d-bde4-3ba0de7d8e3d)
![image](https://github.com/user-attachments/assets/889cc360-0940-4179-b9d5-91abfe395ddd)

After typing all this I tried one last time with a symbol and it gave me the error (always when clicking 'Save Attributes', even when you delete the 'SYMBOL' text in ObjectType. You need to cancel to go on):

![image](https://github.com/user-attachments/assets/b7ced48a-0670-42be-9c77-47db2d7d8210)

```
os: Windows
os_version: 10.0.22621
python_version: 3.11.7
architecture: ('64bit', 'WindowsPE')
machine: AMD64
processor: Intel64 Family 6 Model 158 Stepping 9, GenuineIntel
blender_version: 4.2.1 LTS
bonsai_version: 0.8.0
bonsai_commit_hash: None
last_actions: 
# operator: BIM_OT_load_drawings
# operator: BIM_OT_update_representation
# ifcopenshell.api: geometry.add_representation
>>> {"context": {"cast_type": "entity_instance", "value": 15, "Name": null}, "blender_object": "<bpy_struct, Object(\"IfcAnnotation/Renvooi symbolen\") at 0x000002C7446CD608>", "geometry": "<bpy_struct, Camera(\"15/103\") at 0x000002C7446D1C08>", "coordinate_offset": "None", "total_items": "1", "should_force_faceted_brep": "False", "should_force_triangulation": "False", "should_generate_uvs": "False", "ifc_representation_class": "None", "profile_set_usage": "None", "text_literal": "None"}
# ifcopenshell.api: style.assign_representation_styles
>>> {"shape_representation": {"cast_type": "entity_instance", "value": 206, "Name": null}, "styles": "[]", "should_use_presentation_style_assignment": "False"}
# ifcopenshell.api: geometry.unassign_representation
>>> {"product": {"cast_type": "entity_instance", "value": 66, "Name": "Renvooi symbolen"}, "representation": {"cast_type": "entity_instance", "value": 103, "Name": null}}
# ifcopenshell.api: geometry.remove_representation
>>> {"representation": {"cast_type": "entity_instance", "value": 103, "Name": null}}
# operator: BIM_OT_edit_attributes
# ifcopenshell.api: attribute.edit_attributes
>>> {"product": {"cast_type": "entity_instance", "value": 110, "Name": "so_kl_nuit"}, "attributes": "{'GlobalId': '1nhWjqy4n3Rx9an2WkqBpd', 'Name': 'so_kl_nuit', 'Description': None, 'ObjectType': 'SYMBOL', 'PredefinedType': 'SYMBOL'}"}
last_error: Traceback (most recent call last):
  File "C:\Users\Gebruiker\AppData\Roaming\Blender Foundation\Blender\4.2\extensions\.local\lib\python3.11\site-packages\bonsai\bim\ifc.py", line 408, in execute_ifc_operator
    result = getattr(operator, "_execute")(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Gebruiker\AppData\Roaming\Blender Foundation\Blender\4.2\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\attribute\operator.py", line 109, in _execute
    ifcopenshell.api.run("attribute.edit_attributes", self.file, product=product, attributes=attributes)
  File "C:\Users\Gebruiker\AppData\Roaming\Blender Foundation\Blender\4.2\extensions\.local\lib\python3.11\site-packages\ifcopenshell\api\__init__.py", line 92, in run
    return usecase_function(ifc_file, should_run_listeners=should_run_listeners, **settings)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Gebruiker\AppData\Roaming\Blender Foundation\Blender\4.2\extensions\.local\lib\python3.11\site-packages\ifcopenshell\api\__init__.py", line 252, in wrapper
    result = usecase(*args, **settings)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Gebruiker\AppData\Roaming\Blender Foundation\Blender\4.2\extensions\.local\lib\python3.11\site-packages\ifcopenshell\api\attribute\edit_attributes.py", line 61, in edit_attributes
    if relating_type and relating_type.PredefinedType not in ("NOTDEFINED", None):
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Gebruiker\AppData\Roaming\Blender Foundation\Blender\4.2\extensions\.local\lib\python3.11\site-packages\ifcopenshell\entity_instance.py", line 238, in __getattr__
    raise AttributeError(
AttributeError: entity instance of type 'IFC4X3_ADD2.IfcTypeProduct' has no attribute 'PredefinedType'

ifc: Template-tijdelijk.ifc
schema: IFC4X3
preprocessor_version: IfcOpenShell 0.8.0
originating_system: Bonsai 0.8.0
```

In conclusion I don't know how to always trigger the error, but this is the best I could discover

**Comment by jes-r** [comment]
Cancelling the edit and saving + reloading the exact same file gave me this, as if the problem didn't actually happen. The little circle plus cross is included so it will load the symbol. Different is that both ObjectType and PredefinedType are saying SYMBOL now (as opposed to PredefinedType going to USERDEFINED in the first images).
![image](https://github.com/user-attachments/assets/02637275-02af-448f-906c-8292ee443de1)

**Comment by BIMvoice** [comment]
Thanks @jes-r, root caused and fixed in #8386.

In IFC4X3 IfcAnnotation has a PredefinedType (IfcAnnotationTypeEnum) that IFC4 does not, and that caused the two things you saw.

The save crash: attribute.edit_attributes read the relating type's PredefinedType directly, but several concrete IfcTypeProduct subtypes (IfcFurnishingElementType and similar) have no PredefinedType attribute, so saving a typed annotation raised AttributeError. It now reads it safely.

The empty ObjectType: Bonsai keys the annotation kind off ObjectType, but in IFC4X3 the kinds TEXT, SYMBOL and DIMENSION collide with IfcAnnotationTypeEnum members, so creation put the kind in PredefinedType and left ObjectType empty, which is why the label did not render and you had to type it in by hand. New annotations now set ObjectType to the kind and PredefinedType to USERDEFINED, which is schema valid and keeps the drawing lookups working.

Verified at the ifcopenshell level (the failing edit_attributes call no longer raises and validates clean). I have not driven a live GUI placement, so if you can retest on a recent daily build that would be a helpful confirmation.
